client/state/terms/reducer.js

```diff
@@ -9,15 +9,80 @@ import keyBy from 'lodash/keyBy';
  * Internal dependencies
  */
 import {
+	DESERIALIZE,
 	TERMS_RECEIVE,
-	DESERIALIZE
+	TERMS_REQUEST,
+	TERMS_REQUEST_FAILURE,
+	TERMS_REQUEST_SUCCESS,
+	SERIALIZE
 } from 'state/action-types';
 import { isValidStateWithSchema } from 'state/utils';
+
+import {
+	getSerializedTaxonomyTermsQuery
+} from './utils';
+
 import {
-	termsSchema
+	termsSchema,
+	queriesSchema
 } from './schema';
 
 /**
+ * Returns the updated terms query requesting state after an action has been
+ * dispatched. The state reflects a mapping of serialized query to whether a
+ * network request is in-progress for that query.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function queryRequests( state = {}, action ) {
+	switch ( action.type ) {
+		case TERMS_REQUEST:
+		case TERMS_REQUEST_SUCCESS:
+		case TERMS_REQUEST_FAILURE:
+			const serializedQuery = getSerializedTaxonomyTermsQuery( action.query, action.taxonomy, action.siteId );
```

Comment:
Keep in mind that for posts, we had to support querying from `/me/posts`, which is a large part of why we included the site ID as an optional argument to `getSerializedPostsQuery` for including in the key. Here, since we only query specific to sites, we may just want to structure the state as a nested object, where each key is a site, and values an object of serialized query keys.